iceberg-rust/src/arrow/partition.rs

```diff
@@ -48,7 +48,8 @@ pub async fn partition_record_batches(
     )>,
     ArrowError,
 > {
-    let (partition_sender, partition_reciever): (
+    #[allow(clippy::type_complexity)]
```

Comment:
Disabled clippy for now. I guess this was meant to be rewritten anyway to avoid buffering all the batches, no? Do you mind if I add a todo?